pkg/graph/resolvers/trb_request_form.go

```diff
@@ -42,8 +42,26 @@ func UpdateTRBRequestForm(ctx context.Context, store *storage.Store, input map[s
 		return nil, err
 	}
 
-	if isSubmitted {
+	if isSubmitted && previousStatus != models.TRBFormStatusCompleted {
```

Comment:
Might be useful to have some explanatory commenting about why this is here, to capture the reasoning around handling statuses that we were talking about?